File: 프로그래머스/lv3/64062. 징검다리 건너기/징검다리 건너기.py
def solution(stones, k):
    max_stone = max(stones)
    min_stone = min(stones)
    while min_stone<=max_stone:
        mid =(min_stone+max_stone)//2
        cnt=0
        for i in stones:
            if mid>=i:
                cnt+=1
            else:
                cnt=0
            if cnt>=k:
                max_stone=mid-1
                break
        else:
            min_stone=mid+1
    return min_stone

File: 프로그래머스/3/64062. 징검다리 건너기/징검다리 건너기.java
class Solution {
    public int solution(int[] stones, int k) {
        int st=1;
        int en=200000000+k;
        
        while(st+1<en){
            int mid = (st+en)/2;
            if(canSkip(stones, k, mid)) st = mid;
            else en = mid;
        }
        return st;
    }
    
    public boolean canSkip(int[] stones, int k, int target){
        int count=0;
        for(int stone : stones){
            if(stone<target){
                count++;
            }else{
                count=0;
            }
            if(count==k){
                return false;
            }
        }
        return true;
    }
}